client/packages/purchase_orders/src/DetailView/LineEdit/PurchaseOrderLineEditModal.tsx

```diff
@@ -0,0 +1,126 @@
+import { ModalMode, useDialog, useNotification } from '@common/hooks';
+import { PurchaseOrderFragment } from '../../api';
+import { DialogButton, InlineSpinner } from '@common/components';
+import { useTranslation, Box } from '@openmsupply-client/common';
+import React, { useState } from 'react';
+import { PurchaseOrderLineEdit } from './PurchaseOrderLineEdit';
+import { usePurchaseOrderLine } from '../../api/hooks/usePurchaseOrderLine';
+import { ItemWithStatsFragment } from '@openmsupply-client/system';
+import { createDraftPurchaseOrderLine } from './utils';
+interface PurchaseOrderLineEditModalProps {
+  itemId: string | null;
+  purchaseOrder: PurchaseOrderFragment;
+  mode: ModalMode | null;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const PurchaseOrderLineEditModal = ({
+  itemId,
+  purchaseOrder,
+  mode,
+  isOpen,
+  onClose,
+}: PurchaseOrderLineEditModalProps) => {
+  const t = useTranslation();
+  const { error } = useNotification();
+
+  const lines = purchaseOrder.lines.nodes;
+
+  const [currentItem, setCurrentItem] = useState(
+    lines.find(line => line.item.id === itemId) ?? undefined
+  );
+
+  const {
+    create: { create, isCreating },
+    draft,
+    updatePatch,
+  } = usePurchaseOrderLine(currentItem?.id);
+
+  const onChangeItem = (item: ItemWithStatsFragment) => {
+    const draftLine = createDraftPurchaseOrderLine(item, purchaseOrder.id);
+    item &&
+      updatePatch({
+        ...draftLine,
+      });
+    setCurrentItem({
+      ...draftLine,
+      __typename: 'PurchaseOrderLineNode',
+      item: item,
+    });
+  };
+
+  const handleSave = async () => {
+    try {
+      await create();
+      updatePatch(draft);
+      return true;
+      // TODO add proper error handling by returning type errors from API
```

Comment:
@fergie-nz have you got thoughts about how to handle this? Can you add a follow up issue?

Reply:
yes can do